src/averageAges.js

```diff
@@ -37,7 +42,25 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const mothers = people
+    .map(mother => mother.mother)
+    .map((mother, index, arrayOfMothers) =>
+      arrayOfMothers.indexOf(mother) === index ? mother : null)
+    .filter(mother => mother !== null);
+
+  const mothersAge = people
+    .filter(mothersName => mothers.includes(mothersName.name))
+    .map(year => year.died - year.born);
+
+  const avarageAgeWoman = people
+    .filter(woman => woman.sex === 'f')
+    .map(year => year.died - year.born);
+
+  return withChildren !== undefined
+    ? Math.round((mothersAge
+      .reduce((prev, cur) => prev + cur) / mothersAge.length) * 100) / 100
+    : Math.round((avarageAgeWoman
+      .reduce((prev, cur) => prev + cur) / avarageAgeWoman.length) * 100) / 100;
```

Comment:
first get the array based on the `withChildren` and then calculate the result for that